src/app.py

```diff
@@ -4772,6 +4777,35 @@ def access_level_prefix_dir(dir_name):
     return hm_file_helper.ensureTrailingSlashURL(hm_file_helper.ensureBeginningSlashURL(dir_name))
 
 
+"""
+Check if a user has valid access to update a given entity
+
+Parameters
+----------
+entity : str
+    The entity that is attempting to be updated
+user_token : str 
+    The token passed in via the request header that will be used to authenticate
+
+"""
+
+
+def validate_user_update_privilege(entity, user_token):
+    # A user has update privileges if they are a data admin or are in the same group that registered the entity
+    is_admin = auth_helper_instance.has_data_admin_privs(user_token)
+    if isinstance(is_admin, Response):
+        is_admin = False
+
+    user_write_groups: List[dict] = auth_helper_instance.get_user_write_groups(user_token)
+    if isinstance(user_write_groups, Response):
+        abort_forbidden(f"Entity with ID: {entity['sennet_id']} is not editable without presenting a token.")
+
+    user_group_uuids = [d['uuid'] for d in user_write_groups]
+    if entity['group_uuid'] not in user_group_uuids and is_admin is False:
+        abort_forbidden(f"User does not have write privileges for entity with ID: {entity['sennet_id']}. "
+                        f"Reach out to the help desk to request access to this group.")
```

Comment:
Change `this group` to `['group_name']` would be more explicit in this case since the users will need to tell the help desk which target group they need write access.

Reply:
@yuanzhou This has been fixed.